src/pal/src/misc/time.cpp

```diff
@@ -270,6 +275,29 @@ GetTickCount64()
         }
         retval = (ts.tv_sec * tccSecondsToMillieSeconds)+(ts.tv_nsec / tccMillieSecondsToNanoSeconds);
     }
+#elif HAVE_MACH_CLOCK_MONOTONIC
+    {
+        kern_return_t machRet;
+        clock_serv_t clock;
+        mach_timespec_t mts;
+        host_t host = mach_host_self();
+
+        if((machRet = host_get_clock_service(host, SYSTEM_CLOCK, &clock)) != KERN_SUCCESS)
```

Comment:
@jkotas I also got the results with caching the clock service.
With my changes (no caching) - Average time 3.0 microseconds.
With my changes (with caching) - Average time 0.9 microseconds.
Without my changes (using gettimeofday) - Average time 0.13 microseconds.

There is still an increase in perf cost even with caching enabled.